File: Back-end/RoomSocialBE/DTOs/RoomDTO.cs
using RoomSocialBE.Authentication;
using RoomSocialBE.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace RoomSocialBE.DTOs
{
	public class RoomDTO
	{
		public int id { get; set; }
		public string id_user { get; set; }
		public int id_address { get; set; }
		public int id_categoty { get; set; }
		public string title { get; set; }
		public string? description { get; set; }
		public float arge { get; set; } 
		public float price { get; set; } 
		public int quantity_room { get; set; }
		public string? images { get; set; }
		public DateTime create_day { get; set; } 
		public bool status { get; set; }
		public ApplicationUser user { get; set; }
		public Address? address { get; set; }
		public Category category { get; set; }
	}
}

File: RoomSocialFE/src/app/Models/room.dto.ts
// export interface RoomDTO {
//   id: number;
//   id_user: string;
//   id_adress: number;
//   address: {
//     number_house: number;
//     street_name: string;
//     ward: string;
//     district: string;
//     province: string;
//   };
//   id_category: number; // ID danh mục
//   category?: string;   // Danh mục dưới dạng chuỗi
//   title: string;
//   description: string;
//   arge: number;
//   price: number;
//   quantity_room: number;
//   images: string[];
//   created_day: string;
//   status: boolean;
// }

// Define RoomDTO structure
export interface RoomDTO {
  id: number;
  id_user: number;
  id_address: number;
  id_category: number;
  title: string;
  description: string;
  arge: number;
  price: number;
  quantity_room: number;
  images: string[];
  create_day: string;
  status: boolean;
  user: {
    Id: string;
    UserName: string;
    Email: string;
  };
  address: {
    id: number;
    number_house: string;
    street_name: string;
    ward: string;
    district: string;
    province: string;
  };
  category: {
    id: number;
    name: string;
  };
}
